datalad_service/handlers/objects.py

```diff
@@ -41,23 +40,24 @@ def on_get(self, req, resp, dataset, filekey=None):
                     resp.stream = fd
                     resp.stream_len = os.fstat(fd.fileno()).st_size
                     resp.status = falcon.HTTP_OK
+                # Git objects
                 else:
-                    dir = filekey[:2]
-                    remaining_hex = filekey[2:]
-                    filepath = '.git/objects/{}/{}'.format(dir, remaining_hex)
-                    path = '{}/{}'.format(ds_path, filepath)
-                    compressed_contents = open(path, 'rb').read()
-                    decompressed_contents = zlib.decompress(
-                        compressed_contents)
-                    split_char = b'\x00'
-                    contents = decompressed_contents[decompressed_contents.index(
-                        split_char) + len(split_char):]
-                    resp.body = contents
-                    resp.status = falcon.HTTP_OK
-            except IOError:
-                # File is not kept locally
-                resp.media = {'error': 'file not found'}
-                resp.status = falcon.HTTP_NOT_FOUND
+                    gitCommand = subprocess.run(
+                        ['git', '-C', ds_path, 'cat-file', 'blob', filekey], stdout=subprocess.PIPE)
+                    if (gitCommand.returncode == 0):
+                        resp.body = gitCommand.stdout
+                        resp.status = falcon.HTTP_OK
+                    elif (gitCommand.returncode == 128):
+                        # File is not kept locally
+                        resp.media = {'error': 'file not found'}
+                        resp.status = falcon.HTTP_NOT_FOUND
+                    else:
+                        resp.media = {
+                            'error': 'git object command exited with non-zero'}
+                        resp.status = falcon.HTTP_INTERNAL_SERVER_ERROR
+            except subprocess.CalledProcessError as err:
```

Comment:
I just wondered (for my own education) -- when this exception can be thrown?
according to https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError "raised when a process run by check_call() or check_output() returns a non-zero exit status.", but it is `run` which is used here, not either of those two.
According to coverage report https://codecov.io/gh/OpenNeuroOrg/datalad-service/commit/f112f8045371857e499fd78eb3d3932510a57115#D1-59  this code is not reached at least by tests.

Reply:
Thanks for noticing this, it is a dead exception handler left over from running run() with check=True.